update docs for switch from tokens to API keys (#4313)
PrefectHQ/prefect: docs/orchestration/concepts/api.md

### Authenticating the Client with Cloud <Badge text="Cloud"/>

> If using Prefect Cloud, authentication is required. For interactive use, the most common way to use the Cloud Client is to generate an API key and provide it to the client. After doing so, users can save the key so it persists across all Python sessions:

Reviewer: Same here. Anywhere `key` is used alone should be `API key` IMO.

```python
import prefect

> client = prefect.Client(api_token="YOUR_USER_KEY")

Reviewer 2: Maybe here (and elsewhere) we should call this `"API_KEY"` instead

Author: is it confusing that both user and service accounts are now using API keys? Or do you think there's enough context that you can decide which one should be used?

Reviewer 3: Can we do `API_KEY` and `SERVICE_ACCOUNT_API_KEY` where necessary? It seems like having the default be the user api key and the service account ones be the special case would help with clarity. Service accounts are just special users 🤷

Author: that seems like a good heuristic to follow! 👍

client.save_api_token()
```

Now, starting a client in another session will automatically reload the token:

```python
client = prefect.Client()
assert client._api_token == "API_KEY"  # True
```

> Note that an API key can be provided by environment variable (`PREFECT__CLOUD__AUTH_TOKEN`) or in your Prefect config (under `cloud.auth_token`).

Reviewer 4: I believe it should be `PREFECT__CLOUD__AGENT__AUTH_TOKEN`, right?

> Once provisioned with a user-based API key, the Cloud Client can query for available tenants and login to those tenants. In order to query for tenants, call:

Reviewer 2: ```suggestion
Once provisioned with an API key, the Cloud Client can query for available tenants and login to those tenants. In order to query for tenants, call:
```

Author: same question here re: user vs service account API key, that's definitely why I worded it this way initially

### Authentication <Badge text="Cloud"/>

> The Cloud API also supports direct GraphQL access. While you can still use user-based API keys to access and log in to tenants, you will need to manage the short-lived auth and refresh tokens yourself. Therefore, we recommend using the Python client for user-scoped access. You can make GraphQL queries directly from the client:

Reviewer 2: ```suggestion
The Cloud API also supports direct GraphQL access. While you can still use API keys to access and log in to tenants, you will need to manage the short-lived auth and refresh tokens yourself. Therefore, we recommend using the Python client for user-scoped access. You can make GraphQL queries directly from the client:
```

)
```

> To use keys in programmatic requests, include the key as the authorization header of your GraphQL requests:

Reviewer: Same here.

> To use `TENANT`-scoped keys in programmatic requests, include the key as the authorization header of your GraphQL requests:

Author: this line is still not 100% correct, since it references tenant-scoped. Should it be user or service account?
